spm_funcs.py

```diff
@@ -55,6 +55,12 @@ def get_spm_globals(fname):
         SPM global metric for each 3D volume in the 4D image.
     """
     # +++your code here+++
+    img = nib.load(fname)
+    data = img.get_fdata()
+    spm_vals = np.arange(data.shape[-1])
```

Comment:
Why a range here? This could be dangerous - if an error happens and for whatever reason your program does not break (i.e., an exception badly controlled), you will get an array with actual numbers that you could take for a good output.

Although I don't love the idea of preallocating the returned variable, at least initialize it to zeroes or some special number, or an empty array:
```suggestion
    spm_vals = np.zeros((data.shape[-1], ))
```
```suggestion
    spm_vals = np.empty((data.shape[-1], ))
```

Reply:
This is to say, when coding, you need to anticipate how things may go wrong and make the best effort to provide a reliable solution.